examples/cluster/plot_cluster_comparison.py

```diff
@@ -35,33 +34,46 @@
 from sklearn.preprocessing import StandardScaler
 from itertools import cycle, islice
 
-np.random.seed(0)
 
-# ============
-# Generate datasets. We choose the size big enough to see the scalability
-# of the algorithms, but not too big to avoid too long running times
-# ============
+# %%
+# Generate sample data
+# ---------------------------------
+
+# We choose the size big enough to see the scalability
+# of the algorithms, but not too big to avoid too long running times.
+# Sample data consists of
+# 1. noisy circles 2. noisy moons 3. blobs with varied variance
+# 4. anisotrophic distributions ('stripes') 5. blobs with smaller variance
+# 6. rectangular noise
+
 n_samples = 500
+# 1. make large circle containing smaller circle
 noisy_circles = datasets.make_circles(n_samples=n_samples, factor=0.5, noise=0.05)
+# 2. make two half circles
 noisy_moons = datasets.make_moons(n_samples=n_samples, noise=0.05)
-blobs = datasets.make_blobs(n_samples=n_samples, random_state=8)
-no_structure = np.random.rand(n_samples, 2), None
-
-# Anisotropicly distributed data
-random_state = 170
-X, y = datasets.make_blobs(n_samples=n_samples, random_state=random_state)
+# 3. blobs with varied variances
+rng = np.random.RandomState(0)
+varied = datasets.make_blobs(
+    n_samples=n_samples, cluster_std=[1.0, 2.5, 0.5], random_state=rng
+)
+# 4. Anisotropicly distributed data (stripelike structure)
+X, y = datasets.make_blobs(n_samples=n_samples, random_state=rng)
 transformation = [[0.6, -0.6], [-0.4, 0.8]]
 X_aniso = np.dot(X, transformation)
 aniso = (X_aniso, y)
+# 5. make isotropic Gaussian blobs
+blobs = datasets.make_blobs(n_samples=n_samples, random_state=rng)
+# 6. make random noise
+no_structure = np.random.rand(n_samples, 2), None
 
-# blobs with varied variances
-varied = datasets.make_blobs(
-    n_samples=n_samples, cluster_std=[1.0, 2.5, 0.5], random_state=random_state
-)
-
-# ============
+# %%
 # Set up cluster parameters
-# ============
+# ---------------------------------
+
+# Prepare figure containing
+# 1. circles, 2. moons, 3. Gaussian blobs with variance 4. anisotropic distribution
+# 5. Gaussian blobs without less variance 6. random noise
+
 plt.figure(figsize=(9 * 2 + 3, 13))
```

Comment:
we should move those `matplotlib` next to the figure. I think that we should have a cell only where we only define the parameters.